lib/ai/chat.go

```diff
@@ -136,16 +138,17 @@ func (chat *Chat) Complete(ctx context.Context) (any, error) {
 		},
 	)
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return nil, numTokens, trace.Wrap(err)
 	}
 
 	// fetch the first delta to check for a possible JSON payload
 	var response openai.ChatCompletionStreamResponse
 top:
 	response, err = stream.Recv()
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return nil, numTokens, trace.Wrap(err)
 	}
+	numTokens++
```

Comment:
Why are we adding one to `numTokens`?

Reply:
I was under the impression that every item on the stream (except for `[DONE]`) is a single token. Is the start of the stream also special?

Reply:
I believe you can get an empty delta sometimes, why I am still not sure of. Could you add a check or verify it's always one?